File: src/app/Models/seller.ts
import { User } from "./user";


export class Seller extends User {
    productsListedIds: string[];
    productsSoldIds: string[];

    constructor(
        _id: string, 
        firstName: string,
        lastName: string, 
        email: string, 
        mobileNumber: string, 
        dateOfBirth: Date, 
        password: string,
        role: string,
        productsListedIds: string[],
        productsSoldIds: string[]
    ) {
        super(_id, firstName, lastName, email, password, mobileNumber, dateOfBirth, role);
        this.productsListedIds = productsListedIds;
        this.productsSoldIds = productsSoldIds;
    }
}

File: GoServer/Model/seller.go
package Model

type Seller struct {
	Id        		        string     `json:"_id" bson:"_id"`
	FirstName 		        string     `json:"firstName" bson:"firstName" `
	LastName  		        string     `json:"lastName" bson:"lastName" `
	Email     		        string     `json:"email" bson:"email" `
	Password  		        string     `json:"password" bson:"password" `
	MobileNumber          string     `json:"mobileNumber" bson:"mobileNumber"`
	DateOfBirth           string     `json:"dateOfBirth" bson:"dateOfBirth"`
	Role      		        string     `json:"role" bson:"role"`
	ProductsListedIds     []string   `json:"productsListedIds" bson:"productsListesdIds"`
	ProductsSoldIds       []string   `json:"ProductsSoldIds" bson:"ProductsSoldIds"`
}

